framework/src/wifi_manager/wifi_event_monitor_internal.c

```diff
@@ -0,0 +1,375 @@
+			g_cbk.scan_done(WIFI_UTILS_FAIL, NULL, NULL);
+		}
+		break;
+	case LWNL80211_SCAN_DONE:
+	{
+		if (!g_cbk.scan_done) {
+			break;
+		}
+		wifi_utils_scan_list_s *scan_list = _handle_scan(fd, len);
+		if (scan_list) {
+			g_cbk.scan_done(WIFI_UTILS_SUCCESS, scan_list, 0);
+		} else {
+			g_cbk.scan_done(WIFI_UTILS_FAIL, NULL, NULL);
+		}
+		break;
+	}
+	default:
+		ndbg("Bad status received (%d)\n", msg.status);
+		LWNL_ERR;
+		return -1;
+	}
+	return 0;
+}
+
+static int _wifi_utils_fetch_event(int fd)
+{
+	lwnl80211_cb_status status;
+	uint32_t len;
+	char type_buf[8] = {0,};
+	int nbytes = read(fd, (char *)type_buf, 8);
+
+	if (nbytes < 0) {
+		ndbg("Failed to receive (nbytes=%d, msglen=%d)\n", nbytes, msglen);
+		LWNL_ERR;
+		return -1;
+	}
+
+	memcpy(&status, type_buf, sizeof(lwnl80211_cb_status));
+	memcpy(&len, type_buf + sizeof(lwnl80211_cb_status), sizeof(uint32_t));
+
+	ndbg("%d %d\n", status, len);
+	(void)_wifi_utils_call_event(fd, status, len);
+
+	return 0;
+}
+
+static int _wifi_utils_callback_handler(int argc, char *argv[])
+{
+	LWNL_ENTER;
+	ndbg("run utils callback handler (should be moved to booting)\n");
+	fd_set rfds, ofds;
+
+	int nd = socket(AF_LWNL, SOCK_RAW, LWNL_ROUTE);
+	if (nd < 0) {
+		LWNL_ERR;
+		return -1;
+	}
+
+	int sd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sd < 0) {
+		LWNL_ERR;
+		close(nd);
+		return -1;
+	}
+	struct sockaddr_in saddr;
+	memset(&saddr, 0, sizeof(struct sockaddr));
+	saddr.sin_family = AF_INET;
+	saddr.sin_addr.s_addr = htonl(INADDR_ANY);
+	saddr.sin_port = 9098;
+
+	int res = bind(sd, (struct sockaddr *)&saddr, sizeof(struct sockaddr));
+	if (res < 0) {
+		LWNL_ERR;
+		close(nd);
+		close(sd);
+		return -1;
+	}
+
+	FD_ZERO(&ofds);
+	FD_SET(nd, &ofds);
+	FD_SET(sd, &ofds);
+
+	int maxfd = (nd > sd) ? (nd + 1) : (sd + 1);
+
+	// notify initialization of receive handler is done
+	sem_post(&g_lwnl_signal);
+
+	while (1) {
+		rfds = ofds;
+		res = select(maxfd, &rfds, NULL, NULL, NULL);
+		if (res < 0) {
+			LWNL_ERR;
+			break;
+		}
+
+		if (res == 0) {
+			LWNL_ERR;
+			break;
+		}
+
+		if (FD_ISSET(nd, &rfds)) {
+			// get events from netlink driver
+			res = _wifi_utils_fetch_event(nd);
+			if (res < 0) {
+				ndbg("message currupted\n");
+				break;
+			}
+		} else if (FD_ISSET(sd, &rfds)) {
+			// get terminate event from application
+			nvdbg("get terminate message\n");
+			sem_post(&g_lwnl_signal);
+			break;
+		} else {
+			// unknown error
+			LWNL_ERR;
+			break;
+		}
+	}
+
+	close(nd);
+	close(sd);
+	return 0;
+}
+
+/*
+ * Public
+ */
+void lwnl_start_monitor(void)
+{
+	int tid = task_create("lwnl8021 cb handler", 110, 4096, (main_t)_wifi_utils_callback_handler, NULL);
+	if (tid < 0) {
+		LWNL_ERR;
+	}
+}
+
+void lwnl_stop_monitor(void)
+{
+	_close_cb_handler();
+}
+
+int lwnl_join_monitor(wifi_utils_cb_s *cbk)
+{
+	if (!cbk) {
+		return WIFI_UTILS_INVALID_ARGS;
+	}
+	g_cbk = *cbk;
+
+	return WIFI_UTILS_SUCCESS;
+}
+
+
+int lwnl_leave_monitor(wifi_utils_cb_s *cb)
+{
+	(void)cb;
```

Comment:
How about defining unused or add comment?

Reply:
Attached a comment, please check it. Thanks!